### ISSUE TITLE
data.frame to matrix conversion: Optimize discontinuous dense case

### ISSUE THREAD

**Issue by ksens** [Paradigm4/revealgenomics#99]
@arko-bhattacharya 

Here is test code that highlights the following three different cases for reshaping an expression data.frame to a matrix:

1. sparse  
2. continuous dense
3. discontinuous dense

The goal would be to find a way to use optimizations suggested in this branch by @mmccorry for the discontinuous dense case (which seems to be the majority case at customer A). 

```R
# case 1 - sparse example A
df1 = data.frame(feature_id = c(1, 2), biosample_id = c(3,4), value = c(7, 9))
df1
#    feature_id biosample_id value
# 1:          1            3     7
# 2:          2            4     9
revealgenomics:::convert_data_frame_to_matrix(df1)
# Reshaping expr to Matrix...Handling sparse matrix / discontinuous dense matrix case
# done.
#    3  4
# 1  7 NA
# 2 NA  9

# case 1 - sparse example B
df1 = data.frame(feature_id = c(1:4), biosample_id = c(1:4), value = c(7))
revealgenomics:::convert_data_frame_to_matrix(df1)
# Reshaping expr to Matrix...Handling sparse matrix / discontinuous dense matrix case
# done.
#    1  2  3  4
# 1  7 NA NA NA
# 2 NA  7 NA NA
# 3 NA NA  7 NA
# 4 NA NA NA  7

# case 2 - continuous dense
df1 = data.frame(feature_id = c(1,2,1,2), biosample_id = c(1,1,2,2), value = c(7))
df1
#   feature_id biosample_id value
# 1          1            1     7
# 2          2            1     7
# 3          1            2     7
# 4          2            2     7
revealgenomics:::convert_data_frame_to_matrix(df1)
# Reshaping expr to Matrix...Handling continuous dense matrix case
# done.
# 2 x 2 sparse Matrix of class "dgTMatrix"
#   1 2
# 1 7 7
# 2 7 7

# case 3 - discontinuous dense
df1 = data.frame(feature_id = c(1,2,1,2), biosample_id = c(1,1,3,3), value = c(7))
revealgenomics:::convert_data_frame_to_matrix(df1)
# Reshaping expr to Matrix...Handling sparse matrix / discontinuous dense matrix case
# done.
#   1 3
# 1 7 7
# 2 7 7
```

_Originally posted by @ksens in https://github.com/Paradigm4/revealgenomics/pull/82#issuecomment-519154502_

**Comment by arko-bhattacharya** [comment]
Submitted a pull request with a modified convert_data_frame_to_matrix() function #108. However, if upon benchmarking we find that the performance deteriorates for continuous dense matrix we will have to use the current method along with the prior method by reordering the if conditions.

**Comment by ksens** [comment]
Resolved by #108. Closing issue. 

If there are any further investigations, let us open these up elsewhere.
